utility/src/main/java/org/oppia/android/util/logging/performancemetrics/ProdPerformanceMetricsUtils.kt

```diff
@@ -0,0 +1,88 @@
+package org.oppia.android.util.logging.performancemetrics
+
+import android.app.ActivityManager
+import android.content.Context
+import android.net.TrafficStats
+import org.oppia.android.app.model.OppiaMetricLog
+import org.oppia.android.util.logging.ConsoleLogger
+import java.io.File
+import javax.inject.Inject
+
+/** Utility to extract performance metrics from the underlying Android system. */
+class ProdPerformanceMetricsUtils @Inject constructor(
+  private val context: Context,
+  private val consoleLogger: ConsoleLogger
+) : PerformanceMetricsUtils {
+
+  override fun getApkSize(): Long {
+    var apkSize: Long = 0
+    try {
+      val apkPath =
+        context.packageManager.getPackageInfo(context.packageName, 0).applicationInfo.sourceDir
+      val apkFile = File(apkPath)
+      apkSize = ((apkFile.length() / 1024))
+    } catch (e: Exception) {
+      consoleLogger.e(
+        "PerformanceMetricsUtils",
+        "APK size could not be calculated."
+      )
+    }
+    return apkSize
+  }
+
+  override fun getUsedStorage(): Long {
+    var storageUsage: Long = 0
+    try {
+      val permanentStorageUsage = context.filesDir.totalSpace - context.filesDir.freeSpace
+      val cacheStorageUsage = context.cacheDir.totalSpace - context.cacheDir.freeSpace
+      storageUsage = permanentStorageUsage + cacheStorageUsage
```

Comment:
Is this calculation actually correct? I'd expect totalSpace and freeSpace to be based on the disk, so it'd be the same for both of these. Is that now how those properties work?

Reply:
According to https://developer.android.com/training/data-storage/app-specific#internal-access-files, there are dedicated locations on the internal storage disk for both persistent and cache data. The locations are indicated by cacheDir and filesDir and hence totalSpace and freeSpace properties for both of them won't be the same. Does that help clarify?

Reply:
I think all that means is that there are two directories, nothing more. Per https://developer.android.com/reference/java/io/File#getTotalSpace() totalSpace is the amount of space the _partition_ that holds the file contains, not the directory itself. I suggest testing this with an emulator or real device--I suspect the values will be the same for both directories if you don't have an SD card installed (and maybe even then--I don't think vanilla Android uses the SD card for app storage but I'm not certain).

Reply:
understood -- made amends, PTAL.